DigitalLearningSolutions.Web/Controllers/HomeController.cs

```diff
@@ -1,12 +1,69 @@
-﻿using Microsoft.AspNetCore.Mvc;
-
-namespace DigitalLearningSolutions.Web.Controllers
-{
-    public class HomeController : Controller
-    {
-        public IActionResult Index()
-        {
-            return View();
-        }
-    }
-}
+﻿using Microsoft.AspNetCore.Mvc;
+
+namespace DigitalLearningSolutions.Web.Controllers
+{
+    using System.Collections.Generic;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.Common.MiniHub;
+    using DigitalLearningSolutions.Web.ViewModels.Home;
+    using Microsoft.Extensions.Configuration;
+
+    public class HomeController : Controller
+    {
+        private readonly List<MiniHubSection> Sections = new List<MiniHubSection>(
+            new[]
+            {
+                new MiniHubSection { ControllerName = "Home", ActionName = "Welcome", SectionTitle = "Welcome" },
+                new MiniHubSection { ControllerName = "Home", ActionName = "Products", SectionTitle = "Products" },
+                new MiniHubSection
+                    { ControllerName = "Home", ActionName = "LearningContent", SectionTitle = "Learning Content" },
+            });
+
+        private readonly IConfiguration configuration;
+
+        public HomeController(IConfiguration configuration)
+        {
+            this.configuration = configuration;
+        }
+
+        public IActionResult Index()
+        {
+            return RedirectToAction("Welcome");
+        }
+
+        [HttpGet]
+        [Route("Home/Section/Welcome")]
+        public IActionResult Welcome()
+        {
+            return View(GetLandingPageViewModel(0));
+        }
+
+        [HttpGet]
+        [Route("Home/Section/Products")]
+        public IActionResult Products()
+        {
+            return View(GetLandingPageViewModel(1));
+        }
+
+        [HttpGet]
+        [Route("Home/Section/LearningContent")]
+        public IActionResult LearningContent()
+        {
+            return View(GetLandingPageViewModel(2));
+        }
+
+        private LandingPageViewModel GetLandingPageViewModel(int sectionIndex)

```

Comment:
I'm not massively happy about code which calls this having to know what order the sections are in.
I think I could change MiniHubNavigationModel.Sections to from a list to a Dictionary<T, MiniHubSection> where T : Enumeration, where Enumeration is a base enumeration class as in https://docs.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/enumeration-classes-over-enum-types.

That may be me getting a bit paranoid and excited, so let me know what you think.

Reply:
This sounds really cool, but I think that because of the scope change of the home page, we shouldn't spend ages trying to perfect everything as I think the code is readable/understandable as is.
If you feel like you have some time in hand and that it'll be worth in the long run, feel free to timebox implementing it - I trust your judgement :)

Reply:
May not be worth it if we only ever use a mini-hub pattern here actually. Tech debt ticket? Or you could just keep it in your head in case another mini-hub comes up in the future.